lib/open-amp/resource_table.c

```diff
@@ -72,6 +76,26 @@ static struct fw_resource_table __resource resource_table = {
 		"Zephyr_log",
 	},
 #endif
+
+#if (CONFIG_OPENAMP_RSC_TABLE_VENDOR_RSC_LENGTH > 0)
+	/* Initialize vendor-specific resource entry */
+	.vendor_hdr = {
+		RSC_VENDOR_START,
+		CONFIG_OPENAMP_RSC_TABLE_VENDOR_RSC_LENGTH * sizeof(uint32_t),
+	},
+	/* Populate vendor-specific feature list based on configuration */
+	.vendor_data = {CONFIG_OPENAMP_RSC_TABLE_VENDOR_DATA_FEATURES_0,
+#if (CONFIG_OPENAMP_RSC_TABLE_VENDOR_RSC_LENGTH > 1)
+			, CONFIG_OPENAMP_RSC_TABLE_VENDOR_DATA_FEATURES_1,
+#endif
+#if (CONFIG_OPENAMP_RSC_TABLE_VENDOR_RSC_LENGTH > 2)
+			, CONFIG_OPENAMP_RSC_TABLE_VENDOR_DATA_FEATURES_2,
+#endif
+#if (CONFIG_OPENAMP_RSC_TABLE_VENDOR_RSC_LENGTH > 3)
+			, CONFIG_OPENAMP_RSC_TABLE_VENDOR_DATA_FEATURES_3,
+#endif
+	},
+#endif
```

Comment:
Regarding the implementation I wonder if you should not define your own table in you SoC Adapting this for all vendors could become a nightmare.

This one could be the default one without specific vendor resources 

An example of implementation would be to define a `OPENAMP_VENDOR_ RSC_TABLE` config that would include a `vendor_resource_table.h` instead of defining the `enum rsc_table_entries` and ` struct fw_resource_table  `

Some other/better probably exist...

Reply:
@carlocaione Can you please review since you are the assignee & maintainer?
Thank you!